lib/driver.js

```diff
@@ -232,14 +232,28 @@ class XCUITestDriver extends BaseDriver {
     await this.runReset();
 
     log.info(`Setting up ${this.isRealDevice() ? 'real device' : 'simulator'}`);
-    if (!this.isRealDevice()) {
+
+    if (this.isRealDevice()) {
+      if (this.opts.app) {
+        await this.installApp();
+      }
+    } else {
+      const installAppPromise = new Promise((resolve, reject) => {
```

Comment:
yep, this solution looks better. Updated now.

Please do not merge it yet - I'd like to check e2e test results first.